File: _CodeTopics/LeetCode_contest/weekly/211/211_1.py
class Solution(object):
    def maxLengthBetweenEqualCharacters(self, s):
        """
        :type s: str
        :rtype: int
        """
        
        length = len(s)
        dic = {}
        for i in range(length):
            if dic.has_key(s[i]):
                dic[s[i]].append(i)
            else:
                dic[s[i]] = [i]
        
        res = -1
        for value in dic.values():
            if len(value) >= 2:
                res = max(res, value[-1]-value[0]-1)
        return res
    
"""
https://leetcode-cn.com/submissions/detail/116578019/

54 / 54 个通过测试用例
状态：通过
执行用时: 20 ms
内存消耗: 12.9 MB
"""

File: _CodeTopics/LeetCode/1601-1800/001624--weekly211_1/tran/211_1.cpp
class Solution {
public:
    int maxLengthBetweenEqualCharacters(string s) {

        // 第 211 场周赛第一题
        // https://leetcode-cn.com/submissions/detail/116578019/

        int length = s.size();
        unordered_map<char, vector<int>> dic = {};
        for (int i = 0; i < length; ++i) {
            if ( dic.find(s[i]) != dic.end() ) {
                dic[s[i]].push_back(i);
            } else {
                dic[s[i]] = vector<int>{i};
            }
        }

        int res = -1;
        for (auto& pair : dic) {
            vector<int> vec = pair.second;
            if (vec.size() >= 2) {
                res = max(res, vec.back() - vec[0] - 1);
            }
        }
        return res;
    }
};

/*
https://leetcode.cn/submissions/detail/363823368/

执行用时：
0 ms
, 在所有 C++ 提交中击败了
100.00%
的用户
内存消耗：
6.5 MB
, 在所有 C++ 提交中击败了
5.49%
的用户
通过测试用例：
54 / 54
*/
